pkg/reconciler/clusterbuildtemplate/clusterbuildtemplate.go

```diff
@@ -107,14 +121,41 @@ func (c *Reconciler) Reconcile(ctx context.Context, key string) error {
 	logger := logging.FromContext(ctx)
 
 	// Get the ClusterBuildTemplate resource with this key
-	if _, err := c.clusterBuildTemplatesLister.Get(key); errors.IsNotFound(err) {
+	cbt, err := c.clusterBuildTemplatesLister.Get(key)
+	if errors.IsNotFound(err) {
 		// The ClusterBuildTemplate resource may no longer exist, in which case we stop processing.
 		logger.Errorf("clusterbuildtemplate %q in work queue no longer exists", key)
 		return nil
 	} else if err != nil {
 		return err
 	}
 
-	// TODO: Meaningful reconciliation.
+	if err := c.reconcileImageCaches(ctx, cbt); err != nil {
+		return err
+	}
+
 	return nil
 }
+
+func (c *Reconciler) reconcileImageCaches(ctx context.Context, cbt *v1alpha1.ClusterBuildTemplate) error {
+	ics := resources.MakeImageCaches(cbt)
+
+	eics, err := c.imagesLister.Images(system.Namespace).List(kmeta.MakeVersionLabelSelector(cbt))
+	if err != nil {
+		return err
+	}
+
+	// Make sure we have all of the desired caching resources.
+	if missing := buildtemplate.MissingImageCaches(ics, eics); len(missing) > 0 {
```

Comment:
Taking this even further, `buildtemplate.ReconcileImageCaches` could:

1. Calculate missing images
1. Cache missing images
1. Uncache old images

Reply:
I think I like the split we reach above since the creates can be done in a generic way without the namespace coming into play.  There's already a large number of arguments, and I generally prefer largely single-purpose functions.